sunscreen_runtime/src/runtime.rs

```diff
@@ -392,20 +428,96 @@ where
      * scheme.
      */
     pub fn encrypt<P>(&self, val: P, public_key: &PublicKey) -> Result<Ciphertext>
+    where
+        P: TryIntoPlaintext + TypeName,
+    {
+        self.encrypt_return_components_switched(val, public_key, false)
+            .map(|x| x.ciphertext)
+    }
+
+    #[allow(dead_code)]
+    /**
+     * Encrypts the given [`FheType`](crate::FheType) using the given public
+     * key, and the components used in encrypting the data.
+     *
+     * Returns [`Error::ParameterMismatch`] if the plaintext is incompatible
+     * with this runtime's scheme.
+     */
+    fn encrypt_return_components<P>(
+        &self,
+        val: P,
+        public_key: &PublicKey,
+    ) -> Result<BFVEncryptionComponents>
+    where
+        P: TryIntoPlaintext + TypeName,
+    {
+        self.encrypt_return_components_switched(val, public_key, true)
+    }
+    /**
+     * Encrypts the given [`FheType`](crate::FheType) using the given public
+     * key, returning the encrypted value along with the components added to the
+     * message. See `BFVEncryptionComponents` for the pieces returned.
+     *
+     * Note that this will disable the special modulus!
+     *
+     * Returns [`Error::ParameterMismatch`] if the plaintext is incompatible with this runtime's
+     * scheme.
+     */
+    fn encrypt_return_components_switched<P>(
```

Comment:
The result of this is BFV specific. I believe that is ok for now, but we may want to have a more generic output if we support other backends.